src/capture/middlewares/protobuf_middleware/protobuf_middleware.c

```diff
@@ -0,0 +1,146 @@
+/**
+ * @file
+ * @author Alexandru Mereacre
+ * @date 2022
+ * @copyright
+ * SPDX-FileCopyrightText: © 2022 NQMCyber Ltd and edgesec contributors
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ * @brief File containing the implementation of the protobuf middleware
+ * utilities.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdint.h>
+#include <libgen.h>
+#include <sqlite3.h>
+
+#include "protobuf_encoder.h"
+#include "protobuf_middleware.h"
+
+#include "../../../utils/allocs.h"
+#include "../../../utils/os.h"
+#include "../../../utils/log.h"
+#include "../../../utils/squeue.h"
+#include "../../../utils/eloop.h"
+
+#include "../../pcap_service.h"
+#include "../header_middleware/packet_decoder.h"
+#include "../header_middleware/packet_queue.h"
+
+static const UT_icd tp_list_icd = {sizeof(struct tuple_packet), NULL, NULL,
+                                   free_packet};
```

Comment:
Awesome, you're using UTArray's automatic cleanup functions :)

Reply:
Thanks for the hint :)

There are many places in the code where we need to use this new trick. Will need to do a new PR.